libminifi/include/utils/CollectionUtils.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <set>
+#include <string>
+#include <algorithm>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+bool haveCommonItem(const std::set<std::string>& a, const std::set<std::string>& b) {
```

Comment:
This is specialised for set with no reason. 

In case this just requries two iterables with the same underlying type and you use find instead of count, this is going to work with vectors, lists, etc, while keeping the log(N) efficiency for sets.

Reply:
even though in my opinion "could be made generic" does not imply "should be made generic" (because nobody yet asked for it), since you and @lordgamez also think it is worth generalizing, I am turning this into a template

Reply:
I think the filename itself screams for generalisation. :)

Reply:
so that was that sound!

true, that calling a file `CollectionUtils` and then having a single function in it which only works for `std::set<std::string>` is not great :) 

made it generic, and added tests